### PR TITLE
Fix all DQM paths to properly handle TriggerResults

### PR DESCRIPTION
**Pull request by Dr15Jones**
This pull request fixes all problems from issue #15769 and includes the changes in #15809. The most important additional change on top #15809 changes ConfigBuilder.py and therefore could have a larger impact on the evaluation of this pull request as opposed to #15809.

The ConfigBuilder.py change always makes the dqmoffline*_step path be an EndPath. Previously, based on which processing step is being configured the dqmoffline*_step might sometimes be a Path and other times be an EndPath. The reason appeared to be because sometimes the TriggerResults was being requested by some modules using an empty processName and the hope was one would pick up the HLT results. This change has all modules explicitly state what process name to use when getting the TriggerResults. 

This is needed since the next threaded framework change requires that the consumes of a module and its place on a Path or EndPath guarantee that it can get all the data it requests (or will fail since there is no source or module producing the data it wants). Asking for a TriggerResults for the present process from a Path is therefore not allowed (and was not a good idea before either).

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
hold
I did the full runTheMatrix.py and a few workflows appear to have problems presumably from this change. I'm investigating.

**Comment by Dr15Jones** [comment]
unhold
It was a false alarm. My work area for some reason didn't have everything checked out so the harvesting step on a few workflows failed because a plugin couldn't be found.

**Comment by Dr15Jones** [comment]
Ran the full runTheMatrix.py and all workflows passed.

**Comment by Dr15Jones** [comment]
This fix is needed to allow further work on the threaded framework.

**Comment by Dr15Jones** [comment]
ping

**Comment by Dr15Jones** [comment]
ping

**Comment by Dr15Jones** [comment]
ping
This Ian holding up framework development

**Review by dmitrijus — APPROVED** [review]

**Comment by dmitrijus** [comment]
+1

File: Configuration/Applications/python/ConfigBuilder.py [hunk 1]
```diff
@@ -1939,29 +1941,15 @@ def prepare_DQM(self, sequence = 'DQMOffline'):
...
                         pathName='dqmofflineOnPAT_%d_step'%(i)
 
-		# if both MINIAOD and DQM are run in the same process, schedule DQM in an EndPath
-		if 'PAT' in self.stepMap.keys():
-			# need to put DQM in an EndPath, to access the miniAOD filter results
-			setattr(self.process,pathName, cms.EndPath( getattr(self.process, sequence ) ) )
-		else:
-			# schedule DQM as a standard Path
-			setattr(self.process,pathName, cms.Path( getattr(self.process, sequence) ) )
-		self.schedule.append(getattr(self.process,pathName))
```
Comment (makortel) [inline Configuration/Applications/python/ConfigBuilder.py]: I suspect this removal caused loss of MiniAOD validation histograms in 810pre12, see
https://hypernews.cern.ch/HyperNews/CMS/get/relval/6213/46/2.html

I have a proposal for a fix in #16265.

**Pull request merged**